core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageAutoFeatureStep.java

```diff
@@ -445,13 +452,15 @@ public void addReflectiveMethod(Map<String, ReflectionInfo> reflectiveClasses, R
         }
     }
 
-    public void addReflectiveClass(Map<String, ReflectionInfo> reflectiveClasses, boolean constructors, boolean method,
+    public void addReflectiveClass(Map<String, ReflectionInfo> reflectiveClasses, Set<String> forcedNonWeakClasses,
```

Comment:
I guess we can make this method private?

Reply:
Instead of changing this method and its signature perhaps using the weak parameter is enough?

Reply:
Unfortunately not really, because here we pass in an array of classNames, not just one class